rust/ql/src/queries/security/CWE-918/RequestForgery.rs

```diff
@@ -0,0 +1,39 @@
+// BAD: Endpoint handler that makes requests based on user input
+async fn vulnerable_endpoint_handler(req: Request) -> Result<Response> {
+    // This request is vulnerable to SSRF attacks as the user controls the
+    // entire URL
+    let response = reqwest::get(&req.user_url).await;
+    
+    match response {
+        Ok(resp) => {
+            let body = resp.text().await.unwrap_or_default();
+            Ok(Response {
+                message: "Success".to_string(),
+                data: body,
+            })
+        }
+        Err(_) => Err("Request failed")
+    }
+}
+
+// GOOD: Validate user input against an allowlist
+async fn secure_endpoint_handler(req: Request) -> Result<Response> {
+    // Allow list of specific, known-safe URLs
+    let allowed_hosts = ["api.example.com", "trusted-service.com"];
+    
+    if !allowed_hosts.contains(&req.user_url) {
+        return Err("Untrusted domain");
+    }
+    // This request is safe as the user input has been validated
+    let response = reqwest::get(&req.user_url).await;
+    match response {
+        Ok(resp) => {
+            let body = resp.text().await.unwrap_or_default();
+            Ok(Response {
+                message: "Success".to_string(),
+                data: body,
+            })
+        }
+        Err(_) => Err("Request failed")
+    }
+}
```

Comment:
> Since there's a whole bunch of web frameworks for Rust, I've made the example in the qhelp generic. I think using a specific web framework would be jarring for users who are using a different one.

That seems reasonable, though remember that Autofix uses the documentation and examples as guidance for its fixes - we should keep an eye our for Autofix writing "generic" code that won't compile.  My intuition is that it's close enough to real code that Autofix will figure this out, and it won't be a problem.

Reply:
That's a good point!

I suspect writing the example for a specific framework could also throw Autofix off, if it starts suggesting a fix that uses the wrong framework because we use it in the example. If it turns out to be a problem, we could add something like "the example below shows the generic pattern, the correction should be adapted to the specific web framework used".